python/test/unit/language/test_core.py

```diff
@@ -1147,6 +1147,26 @@ def kernel(X, Z, BITCAST: tl.constexpr):
         assert to_numpy(z_tri) == z_ref
 
 
+@pytest.mark.parametrize("dtype_str, num_warps", [(dtype_str, num_warps) for dtype_str in int_dtypes + float_dtypes for num_warps in [4, 8]])
+def test_cat(dtype_str, num_warps):
+    check_type_supported(dtype_str)
+
+    @triton.jit
+    def kernel(X, Y, Z, N: tl.constexpr):
+        offs = tl.arange(0, N)
+        x = tl.load(X + offs)
+        y = tl.load(Y + offs)
+        z = tl.cat(x, y, can_reorder=True)
+        tl.store(Z + tl.arange(0, 2 * N), z)
+
+    x = torch.randint(0, 10, (128,), dtype=getattr(torch, dtype_str), device='cuda')
+    y = torch.randint(0, 10, (128,), dtype=getattr(torch, dtype_str), device='cuda')
+    z_ref = torch.cat([x, y], dim=0).sum()
+    z = torch.zeros((256,), dtype=getattr(torch, dtype_str), device='cuda')
+    kernel[(1, )](x, y, z, N=128, num_warps=num_warps)
+    assert z.sum() == z_ref
```

Comment:
could you add another test to check that there are no duplicates?

Reply:
Definitely

Reply:
Done